yt_dlp/extractor/chingari.py

```diff
@@ -0,0 +1,199 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import itertools
+import json
+
+from .common import InfoExtractor
+from ..compat import compat_urllib_parse_unquote_plus
+from ..utils import (
+    clean_html,
+    ExtractorError,
+    str_to_int,
+)
+
+
+class ChingariBaseIE(InfoExtractor):
+    def _get_post(self, id, post_data):
+        media_data = post_data['mediaLocation']
+        base_url = media_data['base']
+        author_data = post_data.get('authorData', {})
+        song_data = post_data.get('song', {})  # revist this in future for differentiating b/w 'art' and 'author'
+
+        formats = []
+        for frmt, frmt_path in media_data.get('transcoded', {}).items():
+            formats.append({
+                'format_id': frmt,
+                'width': str_to_int(frmt[1:]),
+                'url': base_url + frmt_path,
+            })
+        if media_data.get('path'):
+            formats.append({
+                'format_id': 'original',
+                'format_note': 'Direct video.',
+                'url': base_url + '/apipublic' + media_data['path'],
+                'quality': 10,
+            })
+        self._sort_formats(formats)
+
+        return {
+            'id': id,
+            'title': compat_urllib_parse_unquote_plus(clean_html(post_data.get('caption'))),
+            'description': compat_urllib_parse_unquote_plus(clean_html(post_data.get('caption'))),
+            'duration': media_data.get('duration'),
+            'thumbnail': base_url + media_data.get('thumbnail', ''),
+            'like_count': post_data.get('likeCount'),
+            'view_count': post_data.get('viewsCount'),
+            'comment_count': post_data.get('commentCount'),
+            'repost_count': post_data.get('shareCount'),
+            'timestamp': str_to_int(post_data.get('created_at')) // 1000,
```

Comment:
Do you have to use `str_to_int`? If not,
```suggestion
            'timestamp': int_or_none(post_data.get('created_at'), 1000),
```
If you do, still take care that `None // 1000` doesnt throw error

Reply:
Taken care of.